api/apps/articles/model.coffee

```diff
@@ -161,24 +163,26 @@ validate = (input, callback) ->
   Joi.validate whitelisted, schema, callback
 
 update = (article, input, callback) ->
-  input.published_at = new Date if input.published and not article.published
-  article = _.extend article, input, updated_at: new Date
-  User.find article.author_id, (err, author) ->
+  input.published_at = new Date if(input.published and not article.published and not input.published_at)
+  User.find (input.author_id or article.author_id), (err, author) ->
     return callback err if err
-    article = addSlug article, author
+    article = _.extend article, input, updated_at: new Date
+    article = addSlug article, input, author
     article = denormalizeAuthor article, author
     callback null, article
 
-addSlug = (article, author, callback) ->
+addSlug = (article, input, author, callback) ->
   titleSlug = _s.slugify(article.title).split('-')[0..7].join('-')
-  if article.slug
-    slug = article.slug
+  article.slugs ?= []
+  if input.slug? and (input.slug != _.last(article.slugs))
+    slug = input.slug
   else if author
     slug = _s.slugify(author.user.name) + '-' + titleSlug
   else
     slug = titleSlug
-  article.slugs ?= []
-  article.slugs.push slug unless slug in article.slugs
+  article.slugs.push slug
+  article.slugs = _.uniq(article.slugs.reverse())
+  article.slugs.reverse()
```

Comment:
Not sure reversing twice here is necessary. Seems like you add the new slug to the end, then reverse the whole thing so it's added to the beginning. Might be more straightforward to just `article.slugs = [slug].concat _.uniq(article.slugs)`.